src/hotspot/cpu/x86/c1_LIRAssembler_x86.cpp

```diff
@@ -3025,31 +3025,31 @@ void LIR_Assembler::shift_op(LIR_Code code, LIR_Opr left, jint count, LIR_Opr de
 void LIR_Assembler::store_parameter(Register r, int offset_from_rsp_in_words) {
   assert(offset_from_rsp_in_words >= 0, "invalid offset from rsp");
   int offset_from_rsp_in_bytes = offset_from_rsp_in_words * BytesPerWord;
-  assert(offset_from_rsp_in_bytes < frame_map()->reserved_argument_area_size(), "invalid offset");
+  assert(offset_from_rsp_in_bytes <= frame_map()->reserved_argument_area_size(), "invalid offset");
```

Comment:
I don't understand this change.  Doesn't the byte range go from 0 to frame_map()->reserved_argument_area_size() - 1?

Reply:
It's really confusing here, I'm sure `reserved_argument_area_size` contains extra stubs for jsr292, but not sure if other stub slots (i.e. CounterOverflowStub) are included.

Your suggestion`Maybe there is a way to calculate actual frame size required based on actual stores emitted` should be correct and I'm trying to see if it works.